src/migration/1749656464598-SourceMemberFollowers.ts

```diff
@@ -0,0 +1,105 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class SourceMemberFollowers1749656464598 implements MigrationInterface {
+    name = 'SourceMemberFollowers1749656464598'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`DROP TRIGGER IF EXISTS increment_squad_members_count ON "source_member"`);
+        await queryRunner.query(`DROP FUNCTION IF EXISTS increment_squad_members_count()`);
+        await queryRunner.query(`DROP TRIGGER IF EXISTS blocked_squad_members_count ON "source_member"`);
+        await queryRunner.query(`DROP FUNCTION IF EXISTS blocked_squad_members_count()`);
+        await queryRunner.query(`DROP TRIGGER IF EXISTS decrement_squad_members_count ON "source_member"`);
+        await queryRunner.query(`DROP FUNCTION IF EXISTS decrement_squad_members_count()`);
+        await queryRunner.query(`DROP TRIGGER IF EXISTS removed_squad_members_count ON "source_member"`);
+
+        await queryRunner.query(`
+            CREATE OR REPLACE FUNCTION increment_source_members_count()
+                RETURNS TRIGGER
+                LANGUAGE PLPGSQL
+                AS
+            $$
+            BEGIN
+                UPDATE source
+                SET flags = jsonb_set(
+                              flags,
+                              '{totalMembers}',
+                              to_jsonb(COALESCE(CAST(flags->>'totalMembers' AS INTEGER), 0) + 1)
+                            )
+                WHERE id = NEW."sourceId";
+                RETURN NEW;
+            END;
+            $$
+        `);
+
+        await queryRunner.query(`
+            CREATE OR REPLACE TRIGGER increment_source_members_count
+            AFTER INSERT ON "content_preference"
+            FOR EACH ROW
+            WHEN (NEW.type = 'source' AND NEW.status IN ('follow', 'subscribed'))
+            EXECUTE PROCEDURE increment_source_members_count()
+        `);
+
+        await queryRunner.query(`
+            CREATE OR REPLACE FUNCTION blocked_source_members_count()
+                RETURNS TRIGGER
+                LANGUAGE PLPGSQL
+                AS
+            $$
+            BEGIN
+                UPDATE source
+                SET flags = jsonb_set(
+                              flags,
+                              '{totalMembers}',
+                              to_jsonb(COALESCE(CAST(flags->>'totalMembers' AS INTEGER), 0) - 1)
+                            )
+                WHERE id = NEW."sourceId";
+                RETURN NEW;
+            END;
+            $$
+            `);
+
+        await queryRunner.query(`
+            CREATE OR REPLACE TRIGGER blocked_source_members_count
+            AFTER UPDATE ON "content_preference"
```

Comment:
we also need after update for when row changes from blocked -> follow/subscribed, because user could first block the source and the follow it